compiler/src/iree/compiler/Codegen/SPIRV/Verifiers.cpp

```diff
@@ -324,10 +324,7 @@ LogicalResult verifySPIRVCooperativeMatrixVectorizePassPipeline(
   }
 
   // Check if the C matrix will be promoted for computing shared memory usage.
-  auto matmulResult = cast<linalg::LinalgOp>(op).getDpsInitOperand(0)->get();
-  bool promoteC =
-      !matmulResult.hasOneUse() ||
-      !isa<IREE::Flow::DispatchTensorStoreOp>(*matmulResult.getUsers().begin());
+  bool promoteC = needToPrmoteCForCooperativeMatrix(cast<linalg::LinalgOp>(op));
```

Comment:
on LLVMGPU side we ended up moving this check to the end of the compilation since it is very hard to keep it in sync with the compilation pipeline and there is not much benefit at fail here rather than later as long as it fails explicitly. I would consider doing the same for SPIRV and maybe even share the same pass.

Reply:
+1 makes sense for the verifier, especially if the promotion inference logic in KernelConfig is conservative. One case I could think of in favor of having this here is if the verifier was being used as a source of truth for selecting configs for auto-tuning.

Reply:
You mean the `LLVMGPUCheckIRBeforeLLVMConversion` pass? Yes that's nice. I'll switch to do something similar afterwards. Though I prefer to land this patch first given some of the modifications is still useful. :)

Reply:
yes, sounds good

Reply:
https://github.com/openxla/iree/pull/13684 implemented the above suggestion.